File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/ProtectiveBand.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class ProtectiveBand extends Equipable {
	{
		name = Locale.get("protective_band");
		description = Locale.get("protective_band_desc");
		sprite = "item-black_belt";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).explosionBlock = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).explosionBlock = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ProtectiveBand.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class ProtectiveBand : Equippable() {
	init {
		name = Locale.get("protective_band")
		description = Locale.get("protective_band_desc")
		sprite = "item-black_belt"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.explosionBlock = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.explosionBlock = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}